packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php

```diff
@@ -61,10 +61,11 @@ public function view($id)
      */
     public function printInvoice($id)
     {
-        $invoice = $this->invoiceRepository->findOneWhere([
-            'id'          => $id,
-            'customer_id' => auth()->guard('customer')->id(),
-        ]);
+        $invoice = $this->invoiceRepository->where('id', $id)
```

Comment:
You can add `firstOrFail` and reduce the next three lines check.

Reply:
@devansh-webkul,
Are you saying that the customer doesn't need to check here?

Reply:
~~~php
$invoice = $this->invoiceRepository->where('id', $id)
            ->whereHas('order', function ($query) {
                $query->where('customer_id', auth()->guard('customer')->id());
            })
            ->firstOrFail();
~~~